### ISSUE TITLE
FC 1.4.1 seeks "Clear recent browsing history" permission w/o documentation or explanation

### ISSUE THREAD

**Issue by johnhawkinson** [mozilla/contain-facebook#185]
- Facebook Container Version: 1.3.1 -> 1.4.1
- Operating System + Version: OS X 10.12.6
- Firefox Version: FF 63.0.1

Firefox Container announces an autoupdate (presumably to 1.4.1 per [the version list](https://addons.mozilla.org/en-US/firefox/addon/facebook-container/versions/) but FF doesn't actually say), and the update requires new permissions ("Clear recent browsing history, cookies, and related data"):
<img width="577" alt="screen shot 2018-11-14 at 07 18 08" src="https://user-images.githubusercontent.com/1270718/48482246-2a1b7a80-e7de-11e8-8709-dca5d2d6165f.png">

Examining the documentation ([addons.mozilla.org](https://addons.mozilla.org/en-US/firefox/addon/facebook-container/), github README.md, github issue list, github release notes) don't show anything about this new permission, which leaves me feeling...uncomfortable.

As a user, when an extension update asks for new permissions, I expect the extension documentation to tell me what's going on. Why does it need the new permission? This is especially important for a privacy-oriented extension that is strongly pushed by Mozilla.

It's especially odd because the #132 adds the "storage permission," but that, at least initially, doesn't seem to be what is being sought (maybe it is and the message doesn't really match? Even this is confusing. And "We will need storage permission in a future release," isn't the most convincing justification out there...)

Can you clarify the documentation as to why the new permission is sought and why it is safe to allow it?  

Thank you.

**Comment by groovecoder** [comment]
@jonathanKingston can provide more details, but I believe we need this to better integrate Facebook Container with Multi-Account Containers extension, which we are planning to do soon-ish.

Where would be the best place to put the explanatory text? I'll plan to add it to AMO & GitHub Release notes - anywhere else?

**Comment by SoftVision-PaulOiegas** [comment]
@johnhawkinson yeah, the message seems to be a bit too general. Maybe we can tweak it a little bit?

However, if I remember well from spring (when it was launched), this is how Facebook Container works. It clears the mentioned data for the domains contained, logs you out of them and forces you to relog after they are opened in the Facebook Container.
Yesterday a new release was merged into AMO that contained some extra domains, and I assume the permission was requested for the new domains added.

**Comment by johnhawkinson** [comment]
Thank you! I do see it now (as part of 1.4.2).  Too bad FF doesn't have a clearer mechanism to explain these things (...).

**Comment by JJDLA** [comment]
Sorry.  I'm lost.  Can someone clarify what "Recent History" is being cleared.  Is it just the history from Facebook links or is it today's history?  All saved history?
Thank you

**Comment by johnhawkinson** [comment]
@JJDLA At this time, nothing is being cleared. The permission is granted to the extension in advance of some future changes that might make use of it.

As to what will happen in the future, the intent is it would only affect Facebook history (if that) and not any other kind of browser history. So not today's history, no. And not all saved history, no.

**Comment by JJDLA** [comment]
Thank you for the quick response.  Much appreciated.

**Comment by gordhub** [comment]
> At this time, nothing is being cleared. The permission is granted to the extension in advance of some future changes that might make use of it.

IMHO the current README.md still does not clearly specify the scope of what will be (or might be in the foggy future) cleared. It makes a big difference whether just Facebook-related stuff is cleared, or everything is. And why ask now for a permission that _"might"_ be wanted later? Why not ask for it in the release that _does_ want it?

**Comment by johnhawkinson** [comment]
@groovecoder, I would agree with some of the comments expressed here that the degree of documentation on this, while maybe adequate, doesn't meet the standards of excellence or perfect clarity that one might hope for in a flagship extension for Firefox.

Keep in mind that the people who have the gumption to ask questions in this Github issue are extreme outliers. Most users will blindly accept the choice, and that's behavior we would like to discourage. Advertent users will think about it and maybe try to find the documentation and give up quickly. Only a very small minority of privacy-conscious users will go so far as to find this Github issue and even fewer will register the necessary Github account to comment in it. This extension should be setting the standard for this sort of thing, not doing the bare minimum. (I also suspect that more users will find this issue than will find the Release Notes, but maybe I am wrong about that.)

So, specifically:
* Although the Release Notes have been updated, that information is hard to find. It is not visible when a user goes to `about:addons` and reads the text about the extension. It is not visible when a user follows the "homepage" link at the bottom and is taken to this Github project and shown the README.md. It is not visible when a user somehow painfully navigates to https://addons.mozilla.org/en-US/firefox/addon/facebook-container/.
* In addition to being in all three of those places, it should also be prominently featured. Like at the top. Maybe before any other information about the extension. Because the goal is to make it easily findable by existing users. I'm not sure how long it should be there. That depends on how fast the userbase turns over. A month? Six months? I think there are stats from the Mozilla store as to how many people have elected the upgrade and how many have given up that might inform this decision. Perhaps the ship has unfortunately sailed?

I don't know what the developer appetite for doing all of these thing is. I regret that I was not more outspoken when I leant some affirmation to the current status on Nov. 14.

Happy Thanksgiving, to those of you who celebrate it.

**Comment by johnhawkinson** [comment]
> December

Well, presumably by that time all Firefox users will have seen the extension upgrade prompt and thus we will have missed the window. I think I was ridiculously off when I suggested a month to six months.

Looking now, `extensions.update.interval` is one day, per https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/Updates#Testing_Automatic_Updating, so...I guess the ship has mostly sailed. The 24 hours after 2018-11-13T21:32:53Z were the critical ones.

So that said, I guess the problem remains for the long tail, but we did miss the boat for the major issue.
